sdks/go/README.md

```diff
@@ -17,26 +17,87 @@
     under the License.
 -->
 
-# Go SDK
+# Go SDK Overview
 
 The Apache Beam Go SDK is the Beam Model implemented in the [Go Programming Language](https://go.dev/).
 It is based on the following initial [design](https://s.apache.org/beam-go-sdk-design-rfc).
+Below describes requirements, how to run examples, execute tests, and contribute to the Go SDK.
 
-## How to run the examples
+_A note on Beam specific terminology used in this README._
 
-**Prerequisites**: to use Google Cloud sources and sinks (default for
-most examples), follow the setup
-[here](https://beam.apache.org/documentation/runners/dataflow/). You can
-verify that it works by running the corresponding Java example.
+_This README uses minimally necessary Beam related terminology to help you determine requirements, usage and
+contribution to the Go SDK. A [definitions](#definitions) section below provides short definitions for you to achieve
+these aims._
 
-The examples are normal Go programs and are most easily run directly.
-They are parameterized by Go flags.
-For example, to run wordcount on the Go direct runner do:
+# Requirements
 
+Aside from the obvious [go](https:/go.dev), you will need to clone the
+[Beam Repository](https://github.com/apache/beam) on your local machine.
+
+To keep terminal commands clear in this README, the following is assumed:
+
+```
+export BEAM_ROOT=path/to/where/you/clone/beam/repository
+```
+
+Only required to run examples, execute tests, and contribute to the Go SDK, [git clone](https://git-scm.com/docs/git-clone)
+the Beam repository:
+
+```sh
+git clone https://github.com/apache/beam.git $BEAM_ROOT
 ```
-$ pwd
-[...]/sdks/go
-$ go run examples/wordcount/wordcount.go --output=/tmp/result.txt
+
+or if you [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the Beam repository into your GitHub
+account with `<username>`.
+
+```sh
+git clone git@github.com:<username>/beam.git $BEAM_ROOT
+```
+
+
+# Run Examples
+
+## Additional Requirements
+
+In addition to the [common requirements](#requirements) listed above, the following lists anything additional for
+running most of the [examples in this repository](examples).
+
+### Google Cloud Setup
+
+Most examples require Google Cloud related resources that serve as data [sources](#source) and [sinks](#sink).
+Follow prerequisites listed as setup in https://beam.apache.org/documentation/runners/dataflow for the Java SDK
+(It's optional to run the Java example for validation in that referenced documentation but not required).
+
+## Usage
```

Comment:
An SDK developer shouldn't be informed how to run the examples. The examples are there for users, so it's appropriate to refer to that documentation instead.

The Go SDK quickstart demonstrates running examples:

https://beam.apache.org/get-started/quickstart-go/

I have a task to demonstrate the SDK better for users (because the current quickstart is a bit too quick...)
https://github.com/apache/beam/issues/27300

But someone who works with Go already knows they should be in the Beam Go SDK's go.mod directory. We shouldn't be repeating Go best practices, we should provide the necessary knowledge (where the go.mod file is) but rely on that the users know Go.

You'll notice the bent here is that someone shouldn't be trying to contribute to Beam to learn Go from scratch. We can't teach potential contributors Go. It doesn't scale. (We can help fix and improve their PRs, but we shouldn't be providing documentation that will become stale or be very environment dependant.)